src/Solution/InTempSolution.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpSchool\PhpWorkshop\Solution;
+
+use PhpSchool\PhpWorkshop\Utils\System;
+use Symfony\Component\Filesystem\Filesystem;
+
+class InTempSolution implements SolutionInterface
+{
+    /**
+     * @var string
+     */
+    private $baseDirectory;
+
+    /**
+     * @var string
+     */
+    private $entryPoint;
+
+    /**
+     * @var SolutionFile[]
+     */
+    private $files;
+
+    private function __construct(SolutionInterface $solution)
+    {
+        $fileSystem = new Filesystem();
+
+        $tempDir = System::tempDir();
+        $currentPath = explode('/', System::realpath(__DIR__));
+        $solutionPath = explode('/', System::realpath($solution->getBaseDirectory()));
+        $entryPointPath = explode('/', System::realpath($solution->getEntryPoint()));
+
+        $intersection = array_intersect($currentPath, $solutionPath);
+
+        if (count($intersection) <= 1) {
+            $intersection = explode('/', $tempDir);
+        }
+
+        $basename = implode('/', array_diff($solutionPath, $intersection));
+        $entrypoint = implode('/', array_diff($entryPointPath, $intersection));
```

Comment:
I actually have no idea what this chuck is doing - can you extract it to a method and maybe add a comment or just give it a good name?